src/vtbackend/Screen.cpp

```diff
@@ -3475,7 +3475,11 @@ ApplyResult Screen<Cell>::apply(FunctionDefinition const& function, Sequence con
             _terminal.setTopBottomMargin(decr(seq.param_opt<LineOffset>(0)),
                                          decr(seq.param_opt<LineOffset>(1)));
             break;
-        case DECSTR: _terminal.softReset(); break;
+        case DECSTR:
+            // For VTType VT100 and VT52 ignore this sequence
+            if (_terminal.state().terminalId != VTType::VT100)
+                _terminal.softReset();
```

Comment:
```suggestion
            if (_terminal.state().terminalId == VTType::VT100)
                return ApplyResult::Invalid;
            _terminal.softReset();
```
if should probably be reported (good habbit) if a VT sequence was to be evaluated that shouldn't be supported by that VT type.